### DISCUSSION TITLE
Does @asset cost 1 or 2 cloud credits?

### DISCUSSION THREAD

**Discussion by skrawcz** [dagster-io/dagster#20394] [Q&A] [answered]
Hi

I'm running the following code -- i.e. invoking things from dagster cloud:

```python

from cowsay import main

from dagster import asset
from dagster import AssetExecutionContext
from shared import shared_function

import time
import pandas as pd

@asset
def asset2(context: AssetExecutionContext) -> pd.DataFrame:
    context.log.info(shared_function())
    context.log.info(f"Cowsay version {main.__version__}")
    time.sleep(100)
    context.log.info(f"Finished processing asset2")
    return pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})

@asset
def asset3(asset2: pd.DataFrame):
    # context.log.info(shared_function())
    print(shared_function())
    # context.log.info(f"Cowsay version {main.__version__}")
    time.sleep(300)
    # context.log.info(f"Finished processing asset3 - {asset2.to_string()}")
    print(f"Finished processing asset3 - {asset2.to_string()}")

```

When in the cloud UI I invoke materializations for both of these assets, it costs me 4 cloud credits, instead of 2 as I would expect. Can someone please clarify why it should take 4?

**Comment by cmpadden** [comment]
Hi @skrawcz - I agree that this isn't the most intuitive. In Dagster a credit is used on asset materialization, and on `op` invocation.  From the website:

> We define usage as the sum of asset materializations (when using software-defined assets) and ops (when using ops and jobs) completed by your data pipelines in Dagster. These operations are summed up as ‘credits.’ For example, If you’ve got a pipeline that executes four operations and materializes one asset, that pipeline would consume five credits when it is run. Your monthly bill is based on the total number of credits used across all of your Dagster pipelines.

When using the `@asset` decorator, you are technically running an `op` with an implied `AssetMaterialization` at the end. For that reason, while you have two _assets_, there are 2 credits to going to asset materialization, and 2 credits to ops.

**Reply by skrawcz** [reply]
Thanks for the clarification. In which case I think that paragraph is misleading. The way I'm interpreting:

> If you’ve got a pipeline that executes four operations and materializes one asset

Is that I have a job with four functions annotated with `@op`, `@op`, `@op`, `@op`,  and then one with `@asset` . That totals 5 credits and makes sense. However you just said an `@asset` is two credits, then that text should say 6 credits are used, not 5.

**Reply by cmpadden** [reply, marked as answer]
In this case, that can be interpreted as one of the four ops materialized the asset.

But thank you for the feedback on it not being clear.

**Comment by alexanderfifefd** [comment]
I've used Dagster cloud for two years, been on the new Dagster+ pricing for a couple of months. I distinctly remember reading that quote and having the takeaway that an op or an asset is worth one credit each. 

> We define usage as the sum of asset materializations (when using [software-defined assets](https://docs.dagster.io/concepts/assets/software-defined-assets)) and ops (when using [ops and jobs](https://docs.dagster.io/concepts/ops-jobs-graphs/ops)) completed by your data pipelines in Dagster. These operations are summed up as ‘credits.’ For example, If you've got a pipeline that executes four operations and materializes one asset, that pipeline would consume five credits when it is run. Your monthly bill is based on the total number of credits used across all of your Dagster pipelines.

I even knew that assets ran ops under the hood. But when you state that a pipeline "materializes one asset" I'm not thinking the odd use case where one of the four ops materialise an asset with AssetMaterialization. Or that you are actually talking about three ops and an asset in a roundabout way. In your [asset docs](https://docs.dagster.io/tutorial/writing-your-first-asset) you state 

> To materialize an asset means to create or update it.

Honestly, extremely misleading.
